karaf-features-gen/src/main/groovy/com/github/lburgazzoli/gradle/plugin/KarafFeaturesGenTask.groovy

```diff
@@ -158,39 +167,40 @@ class KarafFeaturesGenTask extends DefaultTask {
      *
      * @param dep The dependency to check
      * @param extension The karafFeatures extension
-     * @param resolvedArtifactMap The map of GAV->ResolvedArtifact
      *
      * @return {@code true} to indicate that the dependency should be wrapped; {@code false} indicates it should not.
      */
-    static boolean shouldWrap(
+    boolean shouldWrap(
             ResolvedComponentResult dep,
-            KarafFeaturesGenTaskExtension extension,
+            Project selectedProject,
             Map<ModuleVersionIdentifier,ResolvedArtifact> resolvedArtifactMap) {
-        return matchesPattern( dep, extension.wraps) || !isOsgi( dep, resolvedArtifactMap )
-    }
+        if ( matchesPattern( dep, extension.wraps ) ) {
+            return true;
+        }
+
+        if ( matchesPattern( dep, "${selectedProject.group}/${selectedProject.name}/${selectedProject.version}") ) {
+            return !hasOsgiManifestHeaders( ( selectedProject.tasks.jar as Jar ).archivePath )
+        }
 
-    /**
-     * Method to determine if a given jar file is am OSGi bundle.
-     * This is useful for determining if we need to wrap it, determined by the existence
-     * of a Bundle-SymbolicName manifest attribute..
-     *
-     * @param dep The dependency to check.
-     * @param resolvedArtifactMap Map of dependency ids (GAV) to ResolvedArtifact
-     *
-     * @return True if this dependency resolved to a jar with an OSGi bundle
-     */
-    static boolean isOsgi(ResolvedComponentResult dep, Map<ModuleVersionIdentifier,ResolvedArtifact> resolvedArtifactMap) {
         ResolvedArtifact resolvedArtifact = resolvedArtifactMap.get( dep.moduleVersion )
-        if ( resolvedArtifact == null ) {
-            // then what?
-            return false;
+        if ( resolvedArtifact != null ) {
+            return !hasOsgiManifestHeaders( resolvedArtifact.file )
         }
 
-        JarFile jarFile = new JarFile( resolvedArtifact.file );
+        return true
+    }
+
+    public boolean hasOsgiManifestHeaders(File file) {
+        JarFile jarFile = new JarFile( file );
         Manifest manifest = jarFile.getManifest();
         if ( manifest != null ) {
-            Object value = manifest.getMainAttributes().getValue("Bundle-SymbolicName")
-            if( value != null && ! value.toString().isEmpty() ) {
+            logger.lifecycle( "Found manifest [${file.absolutePath}], checking for OSGi metadata" )
+            if ( hasAttribute( manifest, "Bundle-SymbolicName" ) ) {
+                logger.lifecycle( "    >> Found Bundle-SymbolicName" )
```

Comment:
Grr, these logger.lifecycle calls should go away.  I used them for debugging.  I will do that in the next pull request,